packages/pigeon/lib/dart_generator.dart

```diff
@@ -403,6 +430,16 @@ void generateDart(DartOptions opt, Root root, StringSink sink) {
     indent.writeln('');
     indent.write('class ${klass.name} ');
     indent.scoped('{', '}', () {
+      // Constructor
```

Comment:
The comments you've added to the methods are incredibly helpful. Unfortunately this new code is in a mega-method that doesn't have that kind of example.

Given that this is a) huge and b) very nested could you start breaking pieces out? This PR doesn't have to do it all retroactively, but the fact that you have a named subsection here suggests this could easily be a new method (with example code), and over time more of the method could be refactored that way.